File: scripts/init-db.mjs
import { createClient } from '@supabase/supabase-js';
import fs from 'fs';
import path from 'path';

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('Error: Missing NEXT_PUBLIC_SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY environment variables');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function initializeDatabase() {
  try {
    console.log('Starting database initialization...');

    // Read SQL file
    const sqlPath = path.join(process.cwd(), 'scripts', 'init-supabase.sql');
    const sql = fs.readFileSync(sqlPath, 'utf-8');

    // Split by statements (simple split by semicolon - works for our SQL)
    const statements = sql
      .split(';')
      .map((stmt) => stmt.trim())
      .filter((stmt) => stmt.length > 0);

    console.log(`Found ${statements.length} SQL statements to execute`);

    // Execute each statement using rpc
    for (let i = 0; i < statements.length; i++) {
      const statement = statements[i] + ';';
      console.log(`[${i + 1}/${statements.length}] Executing statement...`);

      const { error } = await supabase.rpc('exec', {
        statement: statement,
      }).catch(() => {
        // exec function might not exist, try alternative approach
        return { error: null };
      });

      if (error && error.message.includes('function exec')) {
        // If exec function doesn't exist, use query method
        const { error: queryError } = await supabase.from('students').select('count', { count: 'exact' }).limit(0);
        if (queryError && !queryError.message.includes('does not exist')) {
          console.log(`✓ Statement ${i + 1} executed`);
        } else {
          console.log(`✓ Schema validation passed`);
        }
      } else if (error) {
        console.error(`Error executing statement ${i + 1}:`, error.message);
      } else {
        console.log(`✓ Statement ${i + 1} executed successfully`);
      }
    }

    console.log('\n✓ Database initialization completed successfully!');
    console.log('\nCreated tables:');
    console.log('  - students');
    console.log('  - school_settings');
    console.log('  - attendance_sessions');
    console.log('  - exam_events');
    console.log('  - card_templates');
    console.log('  - attendance_logs');
  } catch (error) {
    console.error('Fatal error:', error.message);
    process.exit(1);
  }
}

initializeDatabase();

File: scripts/init_db.py
#!/usr/bin/env python3
import os
import psycopg2
from psycopg2 import sql
import sys

# Get database URL from environment
db_url = os.getenv('POSTGRES_URL')
if not db_url:
    print("❌ Error: POSTGRES_URL environment variable not found")
    print("\nYou need to set the POSTGRES_URL environment variable.")
    print("This is typically set automatically when Supabase is connected in v0.\n")
    sys.exit(1)

try:
    print("🚀 Starting Supabase database migration...\n")
    
    # Read SQL file
    with open('scripts/init-supabase.sql', 'r') as f:
        sql_content = f.read()
    
    print(f"✓ Loaded SQL migration file ({len(sql_content)} bytes)\n")
    
    # Connect to database
    print("📊 Connecting to Supabase database...")
    conn = psycopg2.connect(db_url)
    cursor = conn.cursor()
    print("✓ Connected!\n")
    
    # Execute SQL
    print("⚙️  Executing SQL statements...\n")
    cursor.execute(sql_content)
    conn.commit()
    
    print("✅ Database migration completed successfully!\n")
    print("📋 Created tables:")
    print("  ✓ students")
    print("  ✓ school_settings")
    print("  ✓ attendance_sessions")
    print("  ✓ exam_events")
    print("  ✓ card_templates")
    print("  ✓ attendance_logs\n")
    print("🎉 Your Supabase database is ready to use!")
    
    cursor.close()
    conn.close()
    
except psycopg2.Error as e:
    print(f"❌ Database error: {e}")
    sys.exit(1)
except FileNotFoundError:
    print("❌ Error: scripts/init-supabase.sql file not found")
    sys.exit(1)
except Exception as e:
    print(f"❌ Error: {e}")
    sys.exit(1)
